use_cases/eluc/prescriptors/nsga2/trainer.py

```diff
@@ -38,19 +41,25 @@ def __init__(self,
         self.p_mutation = p_mutation
         self.seed_dir=seed_dir
 
-        # Store eval df if needed
-        if eval_df is not None:
-            self.eval_df = eval_df
-            self.encoded_eval_df = encoder.encode_as_df(eval_df)
-        self.prescriptor = TorchPrescriptor(eval_df, encoder, predictor, batch_size, candidate_params)
+        # Evaluation params
+        self.encoder = encoder
+        self.predictor = predictor
+        self.context_df = eval_df[constants.CAO_MAPPING["context"]]
+        encoded_eval_df = encoder.encode_as_df(eval_df)
+        context_ds = TorchDataset(encoded_eval_df[constants.CAO_MAPPING["context"]].to_numpy(),
+                                  np.zeros((len(encoded_eval_df), len(constants.RECO_COLS))))
+        self.encoded_context_dl = DataLoader(context_ds, batch_size=batch_size, shuffle=False)
+        self.batch_size = batch_size
 
     def _evaluate_candidates(self, candidates: list[Candidate]):
         """
         Calls prescribe and predict on candidates and assigns their metrics to the results.
         """
+        prescriptor_manager = PrescriptorManager(None, self.predictor)
         for candidate in candidates:
-            context_actions_df = self.prescriptor.prescribe(candidate)
-            eluc_df, change_df = self.prescriptor.predict_metrics(context_actions_df)
+            prescriptor = LandUsePrescriptor(candidate, self.encoder, self.batch_size)
+            context_actions_df = prescriptor.torch_prescribe(self.context_df, self.encoded_context_dl)
+            eluc_df, change_df = prescriptor_manager.predict_metrics(context_actions_df)
```

Comment:
This is where we use the new prescriptor logic during evolution. We create a `LandUsePrescriptor` wrapped around whichever `Candidate` we are evaluating. We also create a dummy `PrescriptorManager` that wraps our `Predictor`. Then we can call `torch_prescribe` on the evaluation dataset (which is in the PyTorch format so we don't have to convert) and then predict our metrics using the `PrescriptorManager`.

Reply:
That comment would be useful in the code itself

Reply:
Done!